app/javascript/tests/close_contact/CloseContact.test.js

```diff
@@ -0,0 +1,219 @@
+      } else {
+        emptyCCWrapper.find(Modal.Body).find('Row').at(index).find(iv.inputType).simulate('change', { target: { id: iv.field, value: iv.value } })
+      }
+    })
+
+    emptyCCWrapper.find(Button).at(1).simulate('click');
+    // Once the modal is closed, the values should default back to their nulled out original values
+    expect(emptyCCWrapper.state('showModal')).toBeFalsy();
+    expect(emptyCCWrapper.state('first_name')).toEqual(mockCloseContact1.first_name || '')
+    expect(emptyCCWrapper.state('last_name')).toEqual(mockCloseContact1.last_name || '')
+    expect(emptyCCWrapper.state('primary_telephone')).toEqual(mockCloseContact1.primary_telephone || '')
+    expect(emptyCCWrapper.state('email')).toEqual(mockCloseContact1.email || '')
+    expect(emptyCCWrapper.state('last_date_of_exposure')).toEqual(mockCloseContact1.last_date_of_exposure || null)
+    expect(emptyCCWrapper.state('assigned_user')).toEqual(mockCloseContact1.assigned_user || null)
+    expect(emptyCCWrapper.state('notes')).toEqual(mockCloseContact1.notes || '')
+  });
+
+  it('Properly resets any values in the modal to the default (for existing close contact)', () => {
+    const existingCCWrapper = getShallowWrapper(mockCloseContact2);
+    expect(existingCCWrapper.state('showModal')).toBeFalsy();
+    existingCCWrapper.find(Button).at(0).simulate('click');
+    expect(existingCCWrapper.state('showModal')).toBeTruthy();
+
+    // This iteration assumes the testInputValues order matches the order of the elements in the DOM
+    testInputValues.forEach((iv, index) => {
+      if (iv.inputType === 'DateInput') {
+        existingCCWrapper.find(Modal.Body).find('Row').at(index).find(iv.inputType).simulate('change', iv.value)
+      } else {
+        existingCCWrapper.find(Modal.Body).find('Row').at(index).find(iv.inputType).simulate('change', { target: { id: iv.field, value: iv.value } })
+      }
+    })
+
+    existingCCWrapper.find(Button).at(3).simulate('click');
+    // Once the modal is closed, the values should default back to their nulled out original values
+    expect(existingCCWrapper.state('showModal')).toBeFalsy();
+    expect(existingCCWrapper.state('first_name')).toEqual(mockCloseContact2.first_name || '')
+    expect(existingCCWrapper.state('last_name')).toEqual(mockCloseContact2.last_name || '')
+    expect(existingCCWrapper.state('primary_telephone')).toEqual(mockCloseContact2.primary_telephone || '')
+    expect(existingCCWrapper.state('email')).toEqual(mockCloseContact2.email || '')
+    expect(existingCCWrapper.state('last_date_of_exposure')).toEqual(mockCloseContact2.last_date_of_exposure || null)
+    expect(existingCCWrapper.state('assigned_user')).toEqual(mockCloseContact2.assigned_user || null)
+    expect(existingCCWrapper.state('notes')).toEqual(mockCloseContact2.notes || '')
+  });
+
+  it('Enables submit button when first and last name are not empty', () => {
```

Comment:
Small but it is first _**or**_ last name is empty

Reply:
No I believe it is an `and`. You need both of them.

Reply:
ah this was changed since I last looked